### PR TITLE
ci: add tests and start to move out common code

### PR DESCRIPTION
**Pull request by tjoshum**
There's still plenty to do, but I think there's value in merging this as it is and it'll already make developing further tests easier.

In particular, we need to move the import of the mayastor package to the same basis as the other 'system' nix ones, and find a better way of setting default values for the nodes.

### CODE REVIEW

File: nix/test/mayastorLib/MayastorUtils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+DEFAULT_NEXUS_PORT = "8420"
+DEFAULT_REPLICA_PORT = "8430"
+
+def subsystem_is_discoverable(host, ip, port, subsys):
```
Comment (paulyoong) [inline nix/test/mayastorLib/MayastorUtils.py:4]: Maybe prefix function name with "nvme" as this is nvme specific. We could possibly want to discover other subsystems in the future.

**Comment by gila** [comment]
Nice @tjoshum it is starting to look like something now that can simply be used rather than struggled with. To make it even more agnostic, could you look into the actual test code being outside of the nix file as well? testScript='' <script>'' becomes a testScript = ./actualtest.py; 

this would make reviews in particlar easier in the future.

**Review by paulyoong — APPROVED** [review]
lgtm

**Comment by tjoshum** [comment]
> Nice @tjoshum it is starting to look like something now that can simply use rather than struggled with. To make it even more agnostic, could you look into the actual test code being outside of the nix file as well? testScript='' <script>'' becomes a testScript = ./actualtest.py;
> 
> this would make reviews in particular easier in the future.

Tried with `builtins.readFile ./actualytest.py` and it doesn't allow us to substitute in things like targetIp. Will have to see if there's a way of getting nix to evaluate the string to get the substitutions to work properly.

File: nix/test/mayastorLib/MayastorUtils.py (hunk 1 above)
Comment (tjoshum) [reply nix/test/mayastorLib/MayastorUtils.py]: Even better, I moved it out into an nvmfUtils file

**Comment by tjoshum** [comment]
@gila I'm going to give up on getting the tests in .py files under this PR, in order to get these changes in.

**Comment by tjoshum** [comment]
That said, you probably will be able to define a lot of stuff in the .py files under mayastorUtils.

**Review by tiagolobocastro — APPROVED** [review]

**Pull request merged**